pkgs/build-support/php/hooks/php-script-utils.bash

```diff
@@ -0,0 +1,22 @@
+declare version
+
+setComposeRootVersion() {
+    set +e # Disable exit on error
+
+    if [[ -v version ]]; then
+        echo -e "\e[32mSetting COMPOSER_ROOT_VERSION to $version\e[0m"
+        export COMPOSER_ROOT_VERSION=$version
+    fi
+
+    set -e
+}
+
+hasRequiredPackages() {
+    # Detect if the "require" section of composer has any package to download/install
+    if jq -e 'try( .require | to_entries | map(select(.key | ((startswith("ext-")| not) and (select(. == "php" | not))))) | any )' composer.json ; then
```

Comment:
I don't know anything about those prefixes yet, but yeah we could indeed.

Reply:
I'm starting to think about something. I wonder if it wouldn't be a better idea to introduce a new variable, something like: 'dontComposerInstall' instead of doing an autodetection.
WDYT ?